api/src/main/java/jakarta/enterprise/lang/model/AnnotationTarget.java

```diff
@@ -7,37 +7,98 @@
 import java.util.function.Predicate;
 
 /**
- * Annotation target is anything that can be annotated.
+ * An Annotation target is anything that can be annotated.
  * That is:
  *
  * <ul>
- * <li>a <i>declaration</i>, such as a class, method, field, etc.</li>
- * <li>a <i>type parameter</i>, occuring in class declarations and method declarations</li>
- * <li>a <i>type use</i>, such as a type of method parameter, a type of field, a type argument, etc.</li>
+ * <li>a <i>declaration</i>, such as a class, method, field, etc. See {@link jakarta.enterprise.lang.model.declarations.DeclarationInfo}.</li>
+ * <li>a <i>type parameter</i>, occurring in class declarations and method declarations.</li>
+ * <li>a <i>type use</i>, such as a type of method parameter, a type of field, a type argument, etc. See {@link jakarta.enterprise.lang.model.types.Type}.</li>
  * </ul>
+ *
+ *
+ * @since 4.0.0
  */
 public interface AnnotationTarget {
     // TODO specify equals/hashCode (for the entire .lang.model hierarchy)
     // TODO settle on using Optional everywhere, or allowing null everywhere; it's a mix right now
+    // TODO what about declared vs not declared annotations?
```

Comment:
If we remove `AnnotationInfo.target`, as you propose, will there still be a difference?

Reply:
It relates to inheritance. For example if an annotation has `@Inherited` and is defined on the superclass it should appear in `getAnnotation(..)` methods but not `getDeclaredAnnotation(..)` methods. There appears to be no distinction in this API?

Reply:
Ah gotcha. I originally intended to only return [directly] declared annotations and let users traverse the hierarchy if they also need inherited annotations, but we can certainly revisit that. The `Annotated.getAnnotation[s]` methods from Portable Extensions always return all annotations, including inherited, and don't have a way to only return directly declared annotations. We could move in that direction for sure. (Implementation in Quarkus will become a little more complex, but it's perfectly possible.)

Reply:
Ok yes I see, seems that the core CDI API makes no distinction so we probably don't need to in this API